php/libraries/ImageQcStatus.class.inc

```diff
@@ -0,0 +1,78 @@
+<?php declare(strict_types=1);
+/**
+ * File that describe the Image QC Status class
+ *
+ * PHP Version 7
+ *
+ *  @category Imaging
+ *  @package  Main
+ *  @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS;
+
+/**
+ * This class defines an Image QC Status to allow one to type hint function's
+ * parameters
+ *
+ *  @category Imaging
+ *  @package  Main
+ *  @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris/
+ */
+class ImageQcStatus
+{
+    /**
+     * The label to display to the frontend user
+     *
+     * @var string
+     */
+    private $_qcstatus;
+
+    /**
+     * The selected value
+     *
+     * @var bool
+     */
+    private $_selected;
+
+    /**
+     * Constructor
+     *
+     * @param ?string $qcstatus Either "Pass" or "Fail"
+     * @param ?bool   $selected If selected
+     *
+     * @throws \DomainException When $qcstatus is not one of PASS or FAIL
+     */
+    public function __construct(?string $qcstatus, ?bool $selected)
+    {
+        if (!in_array($qcstatus, ['Pass', 'Fail', null], true)) {
```

Comment:
Not sure if we should do Database validations here. Maybe it's good to have it as a fail early.